src/olympia/files/models.py

```diff
@@ -706,6 +712,50 @@ class Meta:
         db_table = 'validation_annotations'
 
 
+class WebextPermission(ModelBase):
```

Comment:
well `ManyToManyField` isn't great for performance to start with... plus the cost of instantiating the model instances themselves on the python side... and the fact that there can be arbitrary strings for the hosts... all this really pushes me towards storing a simple dumb list.

The only argument I can see for being more granular is that it provides the ability to match files using a given permission easily, but that's also going to be possible with a (true) json field: https://dev.mysql.com/doc/refman/5.7/en/json.html#json-paths , just not from the Django ORM right now.